### DISCUSSION TITLE
failing to checkin docker agw into cloud orchestrstor

### DISCUSSION THREAD

**Discussion by Zawala** [magma/magma#15259] [Q&A] [answered]
Good day @allI have deployed Magma AGW using Docker and all the containers are healthy. The DNS has been set in the control_proxy.yaml file and the control, fluent, and bootstrapper DNS are publicly routable subdomains. The .env file has been configured with the correct paths. However, the AGW is failing to check into the Orchestrator on the cloud and is giving errors.
>Error: <_MultiThreadedRendezvous of RPC that terminated with:
	status = StatusCode.UNAVAILABLE
	details = "failed to connect to all addresses; last error: UNKNOWN: Peer name [state-controller.durihub.co.zw](http://state-controller.durihub.co.zw/) is not in peer certificate"
	debug_error_string = "UNKNOWN:Failed to pick subchannel {created_time:"2023-07-11T11:19:27.981854709+02:00", children:[UNKNOWN:failed to connect to all addresses; last error: UNKNOWN: Peer name [state-controller.durihub.co.zw](http://state-controller.durihub.co.zw/) is not in peer certificate {created_time:"2023-07-11T11:19:27.981846447+02:00", grpc_status:14}]}"
faiiling to forward request, err: failed to connect to all addresses; last error: UNKNOWN: Peer name [dispatcher-controller.durihub.co.zw](http://dispatcher-controller.durihub.co.zw/) is not in peer certificate
ERROR:root:[SyncRPC] gRPC error: failed to connect to all addresses; last error: UNKNOWN: Peer name [dispatcher-controller.durihub.co.zw](http://dispatcher-controller.durihub.co.zw/) is not in peer certificate, reconnecting to cloud.
INFO:root:[SyncRPC] Opening stream to cloudERROR:root:Metrics upload error for service mme (chunk 0)! [StatusCode.UNAVAILABLE] failed to connect to all addresses; last error: UNKNOWN: Peer name [metricsd-controller.durihub.co.zw](http://metricsd-controller.durihub.co.zw/) is not in peer certificate
ERROR:root:Metrics upload error for service enodebd (chunk 0)! [StatusCode.UNAVAILABLE] failed to connect to all addresses; last error: UNKNOWN: Peer name [metricsd-controller.durihub.co.zw](http://metricsd-controller.durihub.co.zw/) is not in peer certificate
ERROR:root:Metrics upload error for service pipelined (chunk 0)! [StatusCode.UNAVAILABLE] failed to connect to all addresses; last error: UNKNOWN: Peer name [metricsd-controller.durihub.co.zw](http://metricsd-controller.durihub.co.zw/) is not in peer certificate
ERROR:root:Metrics upload error for service magmad (chunk 0)! [StatusCode.UNAVAILABLE] failed to connect to all addresses; last error: UNKNOWN: Peer name [metricsd-controller.durihub.co.zw](http://metricsd-controller.durihub.co.zw/) is not in peer certificate
ERROR:root:Metrics upload error for service sessiond (chunk 0)! [StatusCode.UNAVAILABLE] failed to connect to all addresses; last error: UNKNOWN: Peer name [metricsd-controller.durihub.co.zw](http://metricsd-controller.durihub.co.zw/) is not in peer certificate
ERROR:root:Metrics upload error for service state (chunk 0)! [StatusCode.UNAVAILABLE] failed to connect to all addresses; last error: UNKNOWN: Peer name [metricsd-controller.durihub.co.zw](http://metricsd-controller.durihub.co.zw/) is not in peer certificateJul 11 14:39:16 andromeda 39401aebc867[1733]: ERROR:root:Error! Streaming from the cloud failed! [StatusCode.UNAVAILABLE] failed to connect to all addresses; last error: UNKNOWN: Peer name [streamer-controller.durihub.co.zw](http://streamer-controller.durihub.co.zw/) is not in peer certificate
Jul 11 14:39:16 andromeda 39401aebc867[1733]: message repeated 3 times: [ ERROR:root:Error! Streaming from the cloud failed! [StatusCode.UNAVAILABLE] failed to connect to all addresses; last error: UNKNOWN: Peer name [streamer-controller.durihub.co.zw](http://streamer-controller.durihub.co.zw/) is not in peer certificate]<

the dns are appended with terms like streamer, state, dispatcher and metrixd rendering the my set dns on the cloud null and void. Can you help me troubleshoot this issue?Also I have built the fedaratiuon gateway I'm getting the same errors as above on checkin. Kindly assist on how to resolve the errors above

**Comment by mohinali** [comment]
Hello @alli,
Could you please send me your control_proxy.yml file and Also, have you provided the same DNS name in the /etc/hosts file?
Please ensure that you have the orc8r rootCA.pem file located in the /var/opt/magma/certs directory of the Magma.
Thank you!

**Comment by panyogesh** [comment, marked as answer]
Hello @Zawala,
There is one more additional information regarding the certificates. Are the certs being used is generated by Magma or are these self-signed certificate.  In general Magma uses the *.domain kind of certs so individual certs are not required.

If you can check on the steps for certificate generation that will be great.

Thanks
